ProcessLib/HydroMechanics/HydroMechanicsFEM-impl.h

```diff
@@ -215,28 +216,34 @@ void HydroMechanicsLocalAssembler<ShapeFunctionDisplacement,
         auto& eps = _ip_data[ip].eps;
         auto const& sigma_eff = _ip_data[ip].sigma_eff;
 
-        auto const mu = gas_phase.property(MPL::PropertyType::viscosity)
-                            .template value<double>(vars, x_position, t);
-
-        double const K_over_mu =
-            _process_data.intrinsic_permeability(t, x_position)[0] / mu;
+        vars[static_cast<int>(MPL::Variable::phase_pressure)] = N_p.dot(p);
 
-        auto const alpha =
-            solid_phase.property(MPL::PropertyType::biot_coefficient)
-                .template value<double>(vars, x_position, t);
         auto const K_S = solid_material.getBulkModulus(t, x_position);
-        auto const rho_sr = solid_phase.property(MPL::PropertyType::density)
+
+        auto const K = solid.property(MPL::PropertyType::permeability)
+                           .template value<double>(vars, x_position, t);
+        auto const alpha = solid.property(MPL::PropertyType::biot_coefficient)
+                               .template value<double>(vars, x_position, t);
+        auto const rho_sr = solid.property(MPL::PropertyType::density)
                                 .template value<double>(vars, x_position, t);
-        // TODO (FZill) get fluid properties from GPML
-        double const p_fr =
-            (_process_data.fluid_type == FluidType::Fluid_Type::IDEAL_GAS)
-                ? N_p.dot(p)
-                : std::numeric_limits<double>::quiet_NaN();
-        double const rho_fr =
-            _process_data.getFluidDensity(t, x_position, p_fr);
-        double const beta_p = _process_data.getFluidCompressibility(p_fr);
-        auto const porosity = solid_phase.property(MPL::PropertyType::porosity)
+        auto const porosity = solid.property(MPL::PropertyType::porosity)
                                   .template value<double>(vars, x_position, t);
+
+        auto const mu = gas.property(MPL::PropertyType::viscosity)
+                            .template value<double>(vars, x_position, t);
+        auto const rho_fr = gas.property(MPL::PropertyType::density)
+                                .template value<double>(vars, x_position, t);
+        auto const beta_p = gas.property(MPL::PropertyType::density)
+                .template dValue<double>(vars, MPL::Variable::phase_pressure,
+                                         x_position, t) / rho_fr;
+
+        auto const S = gas.hasProperty(MPL::PropertyType::storage)
```

Comment:
let's call it `storage`.

Reply:
okay and I noticed it has to be a medium property